server/stream.go

```diff
@@ -125,13 +127,20 @@ func (a *Account) AddStream(config *StreamConfig) (*Stream, error) {
 
 // AddStreamWithStore adds a stream for the given account with custome store config options.
 func (a *Account) AddStreamWithStore(config *StreamConfig, fsConfig *FileStoreConfig) (*Stream, error) {
+	if strings.HasPrefix(config.Name, mqttStreamNamePrefix) {
```

Comment:
Wonder if we should add in internal streams etc such that these could flow easier and not reply on names?

Reply:
Meaning a non-public `internal` bool in stream config. The consumers could inherit that and bypass some of the subject stuff off of that.

Reply:
As I mentioned in the past, I wanted to make this a configuration but if it is not exported, it won't be persisted (json does not marshall non exported fields, right?). If so, on recovery I won't be able to rely on the config and have to check the name anyway. Of course, in some other places, like here, I could use the internal's config option instead of checking name.